app/src/main/java/com/azuresamples/msalandroidapp/MainActivity.java

```diff
@@ -0,0 +1,552 @@
+    final static String B2C_RESET_SCOPES [] = {"https://fabrikamb2c.onmicrosoft.com/helloapi/demo.read"};
+
+    /* URL of the API we want to request data from */
+    final static String API_URL = "https://fabrikamb2chello.azurewebsites.net/hello";
+
+    final static String SISU_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_SUSI";
+    final static String EDIT_PROFILE_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_edit_profile";
+    final static String RESET_PASSWORD_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_reset";
+
+    /* UI & Debugging Variables */
+    private static final String TAG = MainActivity.class.getSimpleName();
+    Button loginButton;
+    Button editProfileButton;
+    Button signOutButton;
+
+    /* Azure AD Variables */
+    private PublicClientApplication sampleApp;
+    private AuthenticationResult authResult;
+    private String curDomainHint;
+
+    private StringBuilder mLogs;
+
+
+    //
+    // Initiate all our UI, MSAL/B2C
+    //
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        loginButton = (Button) findViewById(R.id.login);
+        editProfileButton = (Button) findViewById(R.id.edit);
+        signOutButton = (Button) findViewById(R.id.clearCache);
+
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                onLoginClicked();
+            }
+        });
+
+        editProfileButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                onEditProfileClicked();
+            }
+        });
+
+        signOutButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                onSignOutClicked();
+            }
+        });
+
+        /* Configure your sample app and save state for this activity */
+        sampleApp = null;
+        if (sampleApp == null) {
+            sampleApp = new PublicClientApplication(
+                    this.getApplicationContext(),
+                    R.raw.b2c_config);
+        }
+
+        /* load in any identity provider we stored */
+        curDomainHint = PreferenceManager
+                .getDefaultSharedPreferences(getBaseContext())
+                .getString("idp", null);
+
+        /* Enable logging */
+        mLogs = new StringBuilder();
+        Logger.getInstance().setLogLevel(Logger.LogLevel.INFO);
+        Logger.getInstance().setEnablePII(true);
+        Logger.getInstance().setEnableLogcatLog(true);
+        Logger.getInstance().setExternalLogger(new ILoggerCallback() {
+            @Override
+            public void log(String tag, Logger.LogLevel logLevel, String message, boolean containsPII) {
+                mLogs.append(message).append('\n');
```

Comment:
Should we also print to Logcat here for a better understanding of the any issues which the customer encounters?